modules/compute/gke-node-pool/variables.tf

```diff
@@ -354,3 +354,9 @@ variable "host_maintenance_interval" {
     error_message = "Invalid host_maintenance_interval value. Must be PERIODIC, AS_NEEDED or the empty string"
   }
 }
+
+variable "initial_node_count" {
+  description = "The initial number of nodes for the pool. In regional clusters, this is the number of nodes per zone. Changing this setting after node pool creation will not make any effect. It cannot be set with static_node_count and must be set to a value between autoscaling_total_min_nodes and autoscaling_total_max_nodes."
```

Comment:
Just to ask a basic question, is initial node count only relevant for the autoscaling case, if yes can we add that  ? 
Also if there is a GKE documentation on how to use this, can we please refer to that ?

Reply:
I did not find any comprehensive documentation about it and that's why I referred to its source code to understand how it works.
[This is the document for api's nodepool object](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters.nodePools#NodePool) and it has some description about initial_node_count.

And regarding if it's bound to autoscaling, I did not find such binding in the source code. But I made this constraint in our module because if static_node_count is not set, we enable autoscaling; and I wanted to have the initial_node_count something between autoscaling range to prevent any possible issue.

```
  dynamic "autoscaling" {
    for_each = local.static_node_set ? [] : [1]
    content {
      total_min_node_count = var.autoscaling_total_min_nodes
      total_max_node_count = var.autoscaling_total_max_nodes
      location_policy      = "ANY"
    }
  }
```